tests/tests/block_tests.cpp

```diff
@@ -1726,4 +1843,90 @@ BOOST_FIXTURE_TEST_CASE( temp_account_balance, database_fixture )
+///
+BOOST_FIXTURE_TEST_CASE( block_size_test, database_fixture )
+{
+   try
+   {
+      ACTORS((alice)(bob));
+
+      const fc::ecc::private_key& key = generate_private_key("null_key");
+      BOOST_TEST_MESSAGE( "Give Alice some money" );
+      transfer(committee_account, alice_id, asset(10000000));
+      generate_block();
+      // generate one more block to initialize fork db
+      db.generate_block( db.get_slot_time(1), db.get_scheduled_witness(1), key, database::skip_nothing );
+
+      const size_t default_block_header_size = fc::raw::pack_size( signed_block_header() );
+      const auto& gpo = db.get_global_properties();
+      const auto block_interval = gpo.parameters.block_interval;
+      idump( (db.head_block_num())(default_block_header_size)(gpo.parameters.maximum_block_size) );
+
+      BOOST_TEST_MESSAGE( "Start" );
+      // Note: a signed transaction with a transfer operation inside is at least 102 bytes;
+      //       after processed, it become 103 bytes;
+      //       an empty block is 112 bytes;
+      //       a block with a transfer is 215 bytes;
+      //       a block with 2 transfers is 318 bytes.
+      for( uint64_t i = 90; i <= 230; ++i )
+      {
+         if( i > 120 && i < 200 ) // skip some
+            i = 200;
+
+         // Temporarily disable undo db and change max block size
+         db._undo_db.disable();
+         db.modify( gpo, [i,&default_block_header_size](global_property_object& p) {
+            p.parameters.maximum_block_size = default_block_header_size + i;
+         });
+         db._undo_db.enable();
+         idump( (i)(gpo.parameters.maximum_block_size) );
+
+         // push a transaction
+         signed_transaction xfer_tx;
+         transfer_operation xfer_op;
+         xfer_op.from = alice_id;
+         xfer_op.to = bob_id;
+         xfer_op.amount = asset(i);
+         xfer_tx.operations.push_back( xfer_op );
+         xfer_tx.set_expiration( db.head_block_time() + fc::seconds( 0x1000 * block_interval ) );
+         xfer_tx.set_reference_block( db.head_block_id() );
+         sign( xfer_tx, alice_private_key );
+         auto processed_tx = PUSH_TX( db, xfer_tx, database::skip_nothing );
+
+         // sign a temporary block
+         signed_block maybe_large_block;
+         maybe_large_block.transactions.push_back(processed_tx);
+         maybe_large_block.previous = db.head_block_id();
+         maybe_large_block.timestamp = db.get_slot_time(1);
+         maybe_large_block.transaction_merkle_root = maybe_large_block.calculate_merkle_root();
+         maybe_large_block.witness = db.get_scheduled_witness(1);
+         maybe_large_block.sign(key);
+         idump( (fc::raw::pack_size(maybe_large_block)) );
+
+         // should fail to push if it's too large
+         if( fc::raw::pack_size(maybe_large_block) > gpo.parameters.maximum_block_size )
+         {
+            BOOST_CHECK_THROW( db.push_block(maybe_large_block), fc::exception );
```

Comment:
The test should ensure that this is triggered a couple of times. Perhaps increment a counter each time and BOOST_CHECK_EQUAL the counter after the loop ends. (Not sure how often it should trigger...)

Reply:
Added a check to make sure it has been triggered at least once. I think it's enough. Didn't use `BOOST_CHECK_EQUAL` because it would be a pain when trying to tweak the loop in the future.